pravega-cli/src/test/java/io/pravega/tools/pravegacli/SecureControllerCommandsTest.java

```diff
@@ -0,0 +1,170 @@
+                .enableTls(tlsEnabled)
+                .restServerPort(restServerPort);
+
+        // Since the server is being built right here, avoiding delegating these conditions to subclasses via factory
+        // methods. This is so that it is easy to see the difference in server configs all in one place. This is also
+        // unlike the ClientConfig preparation which is being delegated to factory methods to make their preparation
+        // explicit in the respective test classes.
+
+        if (authEnabled) {
+            emulatorBuilder.passwdFile("src/test/resources/passwd")
+                    .userName(SecurityConfigDefaults.AUTH_ADMIN_USERNAME)
+                    .passwd(SecurityConfigDefaults.AUTH_ADMIN_PASSWORD);
+        }
+        if (tlsEnabled) {
+            emulatorBuilder.certFile("src/test/resources/server-cert.crt")
+                    .keyFile("src/test/resources/server-key.key")
+                    .jksKeyFile("src/test/resources/server.keystore.jks")
+                    .jksTrustFile("src/test/resources/client.truststore.jks")
+                    .keyPasswordFile("src/test/resources/server.keystore.jks.passwd");
+        }
+
+        localPravega = emulatorBuilder.build();
+
+        // Set the CLI properties.
+        STATE.set(new AdminCommandState());
+        Properties pravegaProperties = new Properties();
+        pravegaProperties.setProperty("cli.controllerRestUri", "localhost:" + restServerPort.toString());
+        pravegaProperties.setProperty("pravegaservice.zkURL", localPravega.getInProcPravegaCluster().getZkUrl());
+        pravegaProperties.setProperty("pravegaservice.containerCount", "4");
+        pravegaProperties.setProperty("cli.authEnabled", "true");
+        pravegaProperties.setProperty("cli.tlsEnabled", "true");
+        pravegaProperties.setProperty("cli.userName", "admin");
+        pravegaProperties.setProperty("cli.password", "1111_aaaa");
+        pravegaProperties.setProperty("cli.security.tls.trustStore.location", "src/test/resources/client.truststore.jks");
+        STATE.get().getConfigBuilder().include(pravegaProperties);
+    }
+
+    ClientConfig prepareValidClientConfig() {
+        return ClientConfig.builder()
+                .controllerURI(URI.create(localPravega.getInProcPravegaCluster().getControllerURI()))
+
+                // TLS-related
+                .trustStore("src/test/resources/ca-cert.crt")
+                .validateHostName(false)
+
+                // Auth-related
+                .credentials(new DefaultCredentials(SecurityConfigDefaults.AUTH_ADMIN_PASSWORD,
+                        SecurityConfigDefaults.AUTH_ADMIN_USERNAME))
+                .build();
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        if (localPravega != null) {
+            localPravega.close();
+        }
+    }
+
+    @Test
+    public void testListScopesCommand() throws Exception {
+        localPravega.start();
```

Comment:
Since each test calls this `start()` method, it can be moved to the test initializer (the method annotated with `@Before`).

Reply:
The `start()` method has been moved to the test initializer.